turtlebot_follower/launch/follower.launch

```diff
@@ -2,29 +2,26 @@
   The turtlebot people (or whatever) follower nodelet.
  -->
 <launch>
-  <arg name="simulation" default="false"/>
+  <arg name="simulation"    default="false"/>
+  <arg name="generate_goal" default="true"/> <!-- Use the default goal point generation -->
+  
+  <!-- Generate Goal: Load camera and centroid goal generation nodelets -->
+  <!-- Else: Assume goal generation is loaded elsewhere -->
+  <include if="$(arg generate_goal)" file="$(find turtlebot_follower)/launch/includes/centroid.launch">
+	<arg name="simulation"  value="$(arg simulation)"/>
+  </include>
+  
+  <node pkg="nodelet" type="nodelet" name="follower_nodelet_manager" args="manager"/>
+  
   <group unless="$(arg simulation)"> <!-- Real robot -->
     <include file="$(find turtlebot_follower)/launch/includes/velocity_smoother.launch.xml">
       <arg name="nodelet_manager"  value="/mobile_base_nodelet_manager"/>
       <arg name="navigation_topic" value="/cmd_vel_mux/input/navi"/>
     </include>
-
-    <include file="$(find turtlebot_bringup)/launch/3dsensor.launch">
-      <arg name="rgb_processing"                  value="true"/>  <!-- only required if we use android client -->
-      <arg name="depth_processing"                value="true"/>
-      <arg name="depth_registered_processing"     value="false"/>
-      <arg name="depth_registration"              value="false"/>
-      <arg name="disparity_processing"            value="false"/>
-      <arg name="disparity_registered_processing" value="false"/>
-      <arg name="scan_processing"                 value="false"/>
-    </include>
   </group>
   <group if="$(arg simulation)">
-    <!-- Load nodelet manager for compatibility -->
-    <node pkg="nodelet" type="nodelet" ns="camera" name="camera_nodelet_manager" args="manager"/>
-
     <include file="$(find turtlebot_follower)/launch/includes/velocity_smoother.launch.xml">
-      <arg name="nodelet_manager"  value="camera/camera_nodelet_manager"/>
+      <arg name="nodelet_manager"  value="/follower_nodelet_manager"/>
```

Comment:
I don't think that this should have a leading slash on the value. That makes it not play well with namespaces.